src/link.js

```diff
@@ -168,11 +167,20 @@ export default class Link extends Plugin {
 			}
 		} );
 
-		// Close on `ESC` press.
-		escPressHandler( {
-			emitter: balloonPanelView,
-			activator: () => balloonPanelView.isVisible,
-			callback: () => this._hidePanel( true )
+		// Focus the form if balloon panel is open and tab key has been pressed.
+		editor.ui.keystrokes.set( 'tab', ( data, cancel ) => {
```

Comment:
> I think that the balloon could have its own keystroke handler. Then it wouldn't have to check whether it's visible and focused or not – that would be automatic.

I changed the code so `LinkFormView` handles the whole thing. 

We need two keystroke handler instances anyway now that `editor.ui.keystrokes` id gone because the balloon must close on esc when:
1. Editable is focused, and that can only be handled by `editor.keystrokes`.
2. A field in the form is focused, which can only be done by a keystroke handler observing all the fields in the balloon.